packages/core/src/sync-historical/index.ts

```diff
@@ -426,12 +429,28 @@ export const createHistoricalSync = async (
     for (const log of logs) {
       if (isLogFactoryMatched({ factory, log })) {
         const address = getChildAddress({ log, factory });
-        if (childAddresses.has(address) === false) {
+        const prevBlockNumber = childAddresses.get(address);
+
+        if (
+          prevBlockNumber === undefined ||
+          prevBlockNumber > hexToNumber(log.blockNumber)
+        ) {
           childAddresses.set(address, hexToNumber(log.blockNumber));
         }
       }
```

Comment:
Nice catch. Is there the same bug in `sync-realtime` as well?

https://github.com/ponder-sh/ponder/blob/fcb838296903b7caf6e84e2236c5a3d752aa15f1/packages/core/src/sync-realtime/index.ts#L548-L556